File: src/main/java/net/overcrave/morpheus/command/MorpheusCommand.java
package net.overcrave.morpheus.command;

import net.overcrave.morpheus.main.Morpheus;
import java.util.ArrayList;
import java.util.List;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabCompleter;

public class MorpheusCommand implements CommandExecutor, TabCompleter
{

	@Override
	public List<String> onTabComplete(CommandSender sender, Command command, String alias, String[] args) 
	{
		if(args.length <= 1)
		{
			List<String> argList = new ArrayList<>();
			argList.add("toggle");
			return argList;
		}
		return null;
	}

	@Override
	public boolean onCommand(CommandSender sender, Command command, String label, String[] args) 
	{
		if(!sender.hasPermission("morpheus.toggle"))
		{
			sender.sendMessage(Morpheus.customConfig.getString("messages.no-permission"));
			return true;
		}
		else if(args.length > 0 && args[0].toLowerCase() == "toggle")
		{
			Morpheus.toggled = !Morpheus.toggled;
			return true;
		}
		else
		{
			sender.sendMessage(Morpheus.customConfig.getString("messages.default"));
			return false;
		}
	}
	
}

File: src/main/kotlin/io/github/commandertvis/morpheus/command/MorpheusCommand.kt
package io.github.commandertvis.morpheus.command

import io.github.commandertvis.morpheus.configuration.Configuration
import io.github.commandertvis.morpheus.plugin
import io.github.commandertvis.morpheus.utility.hasEnoughArgs
import io.github.commandertvis.morpheus.utility.placeholder
import io.github.commandertvis.morpheus.utility.sendFormatMessage
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter

/**
 * The singleton that stores the executor for the main plugin's command.
 */
object MorpheusCommand : CommandExecutor, TabCompleter {

  init {

    with(plugin.getCommand("morpheus")) {
      executor = this@MorpheusCommand
      tabCompleter = this@MorpheusCommand
    }

  }

  override fun onTabComplete(
    sender: CommandSender,
    command: Command,
    alias: String,
    args: Array<out String>
  ): MutableList<String>? {
    if (args.size <= 1) return mutableListOf("toggle")
    return null
  }

  override fun onCommand(
    sender: CommandSender,
    command: Command,
    label: String,
    args: Array<out String>
  ): Boolean {
    if (!sender.hasPermission("morpheus.control")) {
      sender.sendFormatMessage(Configuration.Messages.noPermission)
      return true
    }

    if (args.hasEnoughArgs(1, sender)) return true

    when (args.first().toLowerCase()) {
      "toggle" -> {
        plugin.toggled = !plugin.toggled
        if (plugin.toggled) {
          sender.sendFormatMessage(Configuration.Messages.enabled)
        } else sender.sendFormatMessage(Configuration.Messages.disabled)
      }
      else -> {
        sender.sendFormatMessage(
            Configuration
                .Messages
                .default
                .placeholder("version", plugin.description.version)
        )
      }
    }
    return true
  }
}